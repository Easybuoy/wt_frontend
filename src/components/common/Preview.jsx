import React from "react";
import banner from "../../assets/banner.jpg";
import styled from "styled-components";
const StyledPreview = styled.div`
  display: flex;
  width: 100%;
  height: 100%;
  @media (max-width: 900px) {
    display: none;
  }
  .content {
    display: flex;
    padding: 3rem 3rem;
    align-items: center;
    h2 {
      color: white;
      text-transform: uppercase;
    }

    i {
      color: white;
      padding-right: 1rem;
    }
  }
`;
const Preview = ({ pageName }) => {
  return (
    <StyledPreview>
      <div
        className="view overlay category-detail"
        style={{
          backgroundImage: `url(${banner})`,
          height: "100%",
          width: "100%",
          backgroundRepeat: "no-repeat",
          backgroundSize: "cover",
          backgroundPosition: "center"
        }}
      >
        <div className="content">
          <i className="fas  fa-2x fa-bars"></i>
          <h2>{pageName}</h2>
        </div>
      </div>
    </StyledPreview>
  );
};
export default Preview;
