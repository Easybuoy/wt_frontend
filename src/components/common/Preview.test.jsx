import React from "react";
import { shallow } from "enzyme";

import Preview from "./Preview";

describe("<Preview />", () => {
  let wrapper;

  const props = {
    pagename: "Login"
  };

  beforeEach(() => {
    wrapper = shallow(<Preview {...props} />);
  });

  it("renders the Preview component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
