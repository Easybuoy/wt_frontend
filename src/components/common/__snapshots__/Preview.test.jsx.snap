// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Preview /> renders the Preview component correctly 1`] = `
<styled.div>
  <div
    className="view"
    style={
      Object {
        "backgroundImage": "url(banner.jpg)",
      }
    }
  >
    <div
      className="content"
    >
      <h2 />
    </div>
  </div>
</styled.div>
`;
